crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -323,94 +323,6 @@
 //! # });
 //! ```
 //!
-//! ### Quick refreshing
-//!
-//! A main purpose of [Sliding Sync][MSC] is to provide an API for snappy end
-//! user applications. Long-polling on the other side means that we wait for the
-//! server to respond and that can take quite some time, before sending the next
-//! request with our updates, for example an update in a list's `range`.
-//!
-//! That is a bit unfortunate and leaks through the `stream` API as well. We are
-//! waiting for a `stream.next().await` call before the next request is sent.
-//! The [specification][MSC] on long polling also states, however, that if an
-//! new request is found coming in, the previous one shall be sent out. In
-//! practice that means one can just start a new stream and the old connection
-//! will return immediately — with a proper response though. One just needs to
-//! make sure to not call that stream any further. Additionally, as both
-//! requests are sent with the same positional argument, the server might
-//! respond with data, the client has already processed. This isn't a problem,
-//! the [`SlidingSync`][] will only process new data and skip the processing
-//! even across restarts.
```

Comment:
Maybe some of this is still relevant? Agree that the code example is probably not all that helpful, especially now that we've changed the cancellation situation.

Reply:
Well, there is nothing special to do when cancelling the result of `SlidingSync::stream`. Let's see tomorrow.

Reply:
What about now :-)?